### ISSUE TITLE
shorter chat-id's

### ISSUE THREAD

**Issue by aaannndddyyy** [alnf/toxcore#28]
I think, chat id's should be no longer than normal tox id's.
Is there any reason you'd need more than encrypting and signing, like with normal tox-id's?

**Comment by aaannndddyyy** [comment]
Even after reading through the wiki, I see no reason why a chat would need two different encryption keys. Basically, chatid/groupid should be only the founder's public key or a hash of the chatmanifest. Portentially a checksum and nospam if that makes any sense. but not more.

**Comment by dubslow** [comment]
I agree. The regular core uses only one key for both encrypting and signing, and the public key system is designed so that one key can be used for both. So why have two separate ones for no good reason?

**Comment by alnf** [comment]
@dubslow Can you give me a link to the source code where regular core uses public key signature?

**Comment by aaannndddyyy** [comment]
@alnf even for public key signature you do not need to attach another key. unless you want two different unlinkable identities, but then why combine them in one groupid?.

**Comment by alnf** [comment]
@aaannndddyyy so you think we can use the same one key for authenticated encryption and public signature?

**Comment by aaannndddyyy** [comment]
more or less, yes. But I'd advise to using a common encryption for the entire group. That saves cpu. You encrypt once for all. 

See also #24 which goes in hand with this.

**Comment by alnf** [comment]
@aaannndddyyy what do you mean by more or less? and I did not understand what you mean by encrypt once for all, sorry.

**Comment by aaannndddyyy** [comment]
you can establish a symmetric key for the entire group to use. Then you need not encryp each message n times, but only once with that symmetric key. Then you sign it. But I posted it on the wrong ticket. this really belongs to #24. I was jus reminded because we were talking about the signing here. 
So yes, make groupid short like a standard tox id. make it basically the same with only one indicator for the dht that it is a group and needs to be treated different than a single peer. Or maybe the DHT node will simply know by the way the announce is made. Do not make it arbitrarily long.

**Comment by alnf** [comment]
@aaannndddyyy so, you propose to distribute public and secret keys among every peer in the chat, so everybody can encrypt the messages using these keys and their own? And again, let me ask if you use the same one key for authenticated encryption and public signature?

**Comment by alnf** [comment]
@aaannndddyyy and need not encryp each message n times, but need to sign each message every time?

**Comment by aaannndddyyy** [comment]
i propose DH for #24 and agreeing on a common symm key. Then you only sign the message after encryption. You'd distribute the public key just like you're doing it now. Depending on whether you want it to be otro like or not you either sign only once or n times. But signing only should be faster than encrypting and signing.

**Comment by dubslow** [comment]
@alnf Signatures and their verification are fundamentally a dual-key process: secret key to sign, public key to verify. Theoretically it should be possible to either encrypt or sign something with the same key, but I'll have to check the NaCl docs/existing code.

**Comment by alnf** [comment]
@dubslow so you can't point me out where in the regular core public key signature is used? I mean the link to github with the line number, for example.

**Comment by dubslow** [comment]
Like I said, let me check :)

**Comment by aaannndddyyy** [comment]
in core it uses authenticated encryption. That makes separate signing unnecessary, so I don't think current core is using it that way.

**Comment by aaannndddyyy** [comment]
@alnf : Derive the signing keys from the encryption keys

**Comment by alnf** [comment]
@dubslow "the regular core uses only one key for both encrypting and signing" you said that so I thought you might now right away

**Comment by aaannndddyyy** [comment]
@alnf : with crypto_box you only need one public key per peer. It does encryption AND authentication

**Comment by dubslow** [comment]
@aaannndddyyy crypto_box is not suitable for signed certificates, since it guarantees reputability

**Comment by aaannndddyyy** [comment]
I know. You'd use crypto_sign instead. Afaik, toxcore does not use that. For both, repudiable chat messages and unrepudiable certificates.

**Comment by aaannndddyyy** [comment]
Don't get me wrong: You can have different keypairs, if you need them. Just not as part of the chatid. It is unneeded and makes it very ugly. Under the hood, create and exchange as many key pairs as you like

**Comment by JFreegman** [comment]
@aaannndddyyy  you can't derive the signature key from the encryption key. It only works the opposite way. As for sharing a group secret and doing 1 encryption + signature, this is less secure, far more complicated, and generally the wrong way to do things.

This issue is about shortening chat_id's and that can be done without changing the encryption scheme. We can simply use the chat signature key as the identifier, and peers joining can then derive the encryption key from that. Peer message encryption is entirely separate.

**Comment by aaannndddyyy** [comment]
@JFreegman  yes, you can derive one from the other. Maybe I got the direction wrong. But the point is, you only need one.
And symmetric encryption is insecure? tox currently uses that for 1:1 chats. It does not use public key crypto for that, but a stream cipher. Public key crypto is only used for arriving at a shared secret. So if that is really less secure, 1:1 should change too.
If not, then you can do it in groups as well. But as said, this is another issue. But you would need it for OTR like chat.
Also it is faster.

**Comment by aaannndddyyy** [comment]
It actually also works the other way round, but it may be that this direction is not implemented. 
Here you can see how to derive the encryption keypair from the signing keypair : http://doc.libsodium.org/advanced/ed25519-curve25519.html
As you see, for getting the pk all you need is pk too, no sk. So as soon as you know someone's public ed25519 key you can do both: verify that peer's signature, and encrypt messages to that peer.

**Comment by JFreegman** [comment]
@aaannndddyyy I did not say that symmetric encryption is insecure. I said that sharing a secret amongst group members is. We're not changing the encryption scheme.

**Comment by aaannndddyyy** [comment]
@JFreegman Can you elaborate why it is insecure?

**Comment by JFreegman** [comment]
@aaannndddyyy because it would be trivial for one of the group members to compromise it.

**Comment by landswellsong** [comment]
@JFreegman but the group members are supposed to be close friends

[/trolling]

**Comment by aaannndddyyy** [comment]
I don't understand that. All memebers of the groupchat are supposed to recieve the chat messages of that group. So thewy have the plain text of them already. What would they gain by "compromising" the key?

**Comment by JFreegman** [comment]
@landswellsong true true, my bad!

@aaannndddyyy use your imagination.
